src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    public Ball getRandomBall() {
+        ColorSupplier colorSupplier = new ColorSupplier();
```

Comment:
You are creating a new instance of the `ColorSupplier` class each time the `getRandomBall()` method is called. This is not efficient. Instead, you should create only one instance of `ColorSupplier` and use it every time you need to get a color. You can do this by declaring a class-level `ColorSupplier` variable.